lucille-core/src/main/java/com/kmwllc/lucille/stage/OpenAIEmbedding.java

```diff
@@ -0,0 +1,195 @@
+package com.kmwllc.lucille.stage;
+
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.OpenAIEmbeddingModel;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.knuddels.jtokkit.Encodings;
+import com.knuddels.jtokkit.api.Encoding;
+import com.knuddels.jtokkit.api.EncodingResult;
+import com.knuddels.jtokkit.api.ModelType;
+import com.typesafe.config.Config;
+import dev.langchain4j.data.segment.TextSegment;
+import java.util.ArrayList;
+import java.util.Iterator;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.openai.OpenAiEmbeddingModel;
+import dev.langchain4j.model.output.Response;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This stage uses openAI embedding services to embed a text field in a Lucille document or its children document(s). Will truncate
+ * to token limit before sending request. Retrieves API key from config.
+ *
+ * Config Parameters:
+ * - source (String) : field of which the embedding Stage will retrieve content from
+ * - dest (String, Optional) : name of the field that will hold the embeddings, defaults to "embeddings"
+ * - embed_document (Boolean) : Embeds the document's source if set to true.
+ * - embed_children (Boolean): Embeds the document's children source if set to true.
+ * - api_key (String) : API key used for OpenAI requests
+ * - model_name (String, Optional) : the name of the OpenAI embedding model to use, set default to text-embedding-3-small
+ *    1. text-embedding-3-small
+ *    2. text-embedding-3-large
+ *    3. text-embedding-ada-002
+ *    more details: <a href="https://platform.openai.com/docs/guides/embeddings/embedding-models">...</a>
+ * - dimensions (Integer, Optional) : number of dimensions the resulting embedding should have. Only supported in text-embedding-3
+ * and later models. Default set to null, which will call the model's default dimensions.
+ */
+
+public class OpenAIEmbedding extends Stage {
```

Comment:
most stage names are based on verbs
so I'd suggest OpenAIEmbed or EmbedForOpenAI

Reply:
changed to OpenAIEmbed in [feeaa04](https://github.com/kmwtechnology/lucille/pull/171/commits/feeaa045fb3aa12e3a184b1bfa7564aafdd21d1d)